src/Control/Monad/Logic/Sequence/Compat.hs

```diff
@@ -1,5 +1,27 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE ViewPatterns #-}
 module Control.Monad.Logic.Sequence.Compat
   ( fromSeqT
   , toLogicT
-  , fromLogicT ) where
-import Control.Monad.Logic.Sequence.Internal
+  , fromLogicT
+  , observeT
+  , observe ) where
+
+import Control.Monad.Identity (Identity(..))
+import Control.Monad.Logic.Sequence.Internal hiding ( observeT, observe )
+
+#if !MIN_VERSION_base(4,13,0)
+observeT :: Monad m => SeqT m a -> m a
+#else
+observeT :: MonadFail m => SeqT m a -> m a
+#endif
+observeT (toView -> m) = m >>= go where
+  go (a :< _) = return a
+  go Empty = fail "No results."
+{-# INLINE observeT #-}
+
+observe :: Seq a -> a
+observe (toView -> m) = case runIdentity m of
+  a :< _ -> a
+  Empty -> error "No results."
```

Comment:
Here I think the `HasCallStack` on `error` will include the function name, so maybe you don't need to do so manually?

Reply:
```
*Control.Monad.Logic.Sequence Control.Monad.Logic.Sequence.Compat Control.Applicative λ> Control.Monad.Logic.Sequence.Compat.observe empty
*** Exception: No results.
CallStack (from HasCallStack):
  error, called at src/Control/Monad/Logic/Sequence/Compat.hs:26:12 in logict-sequence-0.1.0.2-inplace:Control.Monad.Logic.Sequence.Compat
  ```

Looks like we're good.